File: code/0013/13.找出数组中重复的数字.cpp
class Solution {
public:
    int duplicateInArray(vector<int>& nums) {
        int n = nums.size();
        for (int num: nums)
            if (num < 0 || num > n - 1)
                return -1;
        for (int i = 0; i < n; i++) {
            while (nums[nums[i]] != nums[i]) 
                swap(nums[i], nums[nums[i]]);
            if (nums[i] != i)
                return nums[i];
        }
        
        return -1;
    }
};

File: code/0013/13.找出数组中重复的数字.py
class Solution(object):
    def duplicateInArray(self, nums):
        """
        :type nums: List[int]
        :rtype int
        """
        n = len(nums)
        for num in nums:
            if num < 0 or num > n - 1:
                return -1
        
        for i in range(n):
            while nums[nums[i]] != nums[i]:
                nums[nums[i]], nums[i] = nums[i], nums[nums[i]]
            if nums[i] != i:
                return nums[i]
        
        return -1
